config/fcos/v1_6_exp/validate.go

```diff
@@ -51,11 +53,28 @@ func (conf Config) Validate(c path.ContextPath) (r report.Report) {
 func (d BootDevice) Validate(c path.ContextPath) (r report.Report) {
 	if d.Layout != nil {
 		switch *d.Layout {
-		case "aarch64", "ppc64le", "x86_64":
+		case "aarch64", "ppc64le", "x86_64", "s390x-eckd", "s390x-virt", "s390x-zfcp":
 		default:
 			r.AddOnError(c.Append("layout"), common.ErrUnknownBootDeviceLayout)
 		}
 	}
+	//Validate s390x layout device specific luks.s390x-device.
+	//s390x layout does not support Mirror.
+	//Validate the s390x layout specific device
+	if d.Layout != nil && (*d.Layout == "s390x-zfcp" ||  *d.Layout == "s390x-eckd") {
+		if util.NilOrEmpty(d.Luks.Device) {
+			r.AddOnError(c.Append(*d.Layout), common.ErrNoLuksBootDevice)
+		}
+		if len(d.Mirror.Devices) > 0 {
+			r.AddOnError(c.Append(*d.Layout), common.ErrMirrorNotSupport)
+		}
+		if *d.Layout == "s390x-zfcp" && util.NotEmpty(d.Luks.Device) && !sdRe.MatchString(*d.Luks.Device) {
```

Comment:
Oh Ok. That condition is not required. Will remove that.

Reply:
The reason for adding two checks to catch the specific error.

If the user specified the layout and forgot to provide the device key ie  luks.s390x-device:. or If the user specified the layout and specified the device key s390x-device and not the value eg /dev/dasda.
why because i created a test binary and tried to test both case with single check causes runtime panic error because
first case it verify that util.NilOrEmpty(d.Luks.Device) this verifies whether the value is nil or empty.

and i used util.NotEmpty guarantee that regex.MustCompile does not run on empty memory address.